### PR TITLE
fix: Retain last selected tab on debugger and user selected filter condition

### LINKED ISSUE #23108: [Bug]: Retain last selected tab on debugger and user selected filter condition
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I want to look at the logs of a js function run, the debugger always switches to the `Response` tab. This is the first annoyance. Next, after selecting the `Logs` tab, I should again select `Console logs` instead of `All logs`. This is the second annoyance. 

Having to do this repeatedly causes an accrual of annoyance :-D

### Steps To Reproduce

https://user-images.githubusercontent.com/94514895/237005961-d454f10d-bab8-401e-9445-0885db2c1529.mov

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by Druthi**
## Description
fix: Retain last selected tab on debugger and user selected filter condition

#### PR fixes following issue(s)
Fixes #23108

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

>
>
#### How Has This Been Tested?
- [x] Manual
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [x] We had an implementation alignment call with stakeholders post QA Round 2
- [x] Cypress test cases have been added and approved by SDET/manual QA
- [x] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM, small nit

**Comment by Druthi** [comment]
> @Druthi User selected filter is updated on selection of the response tab or error tab, As per my understanding filter should not be updated

The filter seems to be selected by default on production as well. I haven't made any changes there

**Comment by AnandiKulkarni** [comment]
@Druthi could you check this behavior - when filter is changed for on jsobject its changed for others in that page, it should not be this
[screen-capture (2).webm](https://github.com/appsmithorg/appsmith/assets/80756091/d591690b-8d47-4bf1-8130-9815eaf753dc)

**Comment by Druthi** [comment]
> @Druthi could you check this behavior - when filter is changed for on jsobject its changed for others in that page, it should not be this [screen-capture (2).webm](https://github.com/appsmithorg/appsmith/assets/80756091/d591690b-8d47-4bf1-8130-9815eaf753dc)

Heya! Checked with @bharath31 and this is an expected behavior. There is only 1 context to the debugger for all entities. Only the last selected context persists but each entity cannot have its on debugger context

File: app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/EntityBottomBar_spec.ts [hunk 1]
```diff
@@ -21,19 +21,13 @@ describe("Entity bottom bar", () => {
     _.jsEditor.CreateJSObject(` return "hello world";`);
     //Verify if bottom bar opens JSEditor.
     _.debuggerHelper.AssertOpen(PageType.JsEditor);
-    // Verify if selected tab is response.
-    _.debuggerHelper.AssertSelectedTab("Response");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/EntityBottomBar_spec.ts:25]: Why is this validation removed?

Comment (arunvjn) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/EntityBottomBar_spec.ts]: Added it back

File: app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/EntityBottomBar_spec.ts [hunk 2]
```diff
@@ -56,8 +55,6 @@ describe("Entity bottom bar", () => {
     _.apiPage.RunAPI();
     _.agHelper.Sleep(1000);
     _.debuggerHelper.AssertOpen(PageType.API);
-    //verify if response tab is selected on execution api.
-    _.debuggerHelper.AssertSelectedTab("Response");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/EntityBottomBar_spec.ts:60]: Can this removal also be brought back?

Comment (arunvjn) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/EntityBottomBar_spec.ts]: I have modified this test to accommodate the new behaviour, where the selected tab is preserved. I have added comments as well

**Pull request merged**